tests/integration_test/shadow_system_test.c

```diff
@@ -0,0 +1,942 @@
+    pSubscriptionList[ 0 ].pTopicFilter = pTopic;
+    pSubscriptionList[ 0 ].topicFilterLength = topicLength;
+
+    /* Generate packet identifier for the UNSUBSCRIBE packet. */
+    globalUnsubscribePacketIdentifier = MQTT_GetPacketId( pContext );
+
+    /* Send UNSUBSCRIBE packet. */
+    mqttStatus = MQTT_Unsubscribe( pContext,
+                                   pSubscriptionList,
+                                   sizeof( pSubscriptionList ) / sizeof( MQTTSubscribeInfo_t ),
+                                   globalUnsubscribePacketIdentifier );
+
+    if( mqttStatus == MQTTSuccess )
+    {
+        mqttStatus = MQTT_ProcessLoop( pContext, MQTT_PROCESS_LOOP_TIMEOUT_MS );
+    }
+
+    if( mqttStatus == MQTTSuccess )
+    {
+        TEST_ASSERT_TRUE( receivedUnsubAck );
+    }
+
+    return mqttStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+static MQTTStatus_t publishToTopic( MQTTContext_t * pContext,
+                                    const char * pTopic,
+                                    const char * pPayload,
+                                    MQTTQoS_t qos )
+{
+    MQTTStatus_t mqttStatus = MQTTSuccess;
+    MQTTPublishInfo_t publishInfo;
+
+    assert( pContext != NULL );
+
+    /* Set the retain flag to false to avoid side-effects across test runs. */
+    publishInfo.retain = false;
+
+    publishInfo.qos = qos;
+    publishInfo.dup = false;
+    publishInfo.pTopicName = pTopic;
+    publishInfo.topicNameLength = strlen( pTopic );
+    publishInfo.pPayload = pPayload;
+    publishInfo.payloadLength = strlen( pPayload );
+
+    /* Get a new packet id. */
+    globalPublishPacketIdentifier = MQTT_GetPacketId( pContext );
+
+    /* Send PUBLISH packet. */
+    mqttStatus = MQTT_Publish( pContext,
+                               &publishInfo,
+                               globalPublishPacketIdentifier );
+
+    if( mqttStatus == MQTTSuccess )
+    {
+        mqttStatus = MQTT_ProcessLoop( pContext, MQTT_PROCESS_LOOP_TIMEOUT_MS );
+    }
+
+    if( mqttStatus == MQTTSuccess )
+    {
+        /* We do not expect a PUBACK from the broker for the QoS 0 PUBLISH. */
+        TEST_ASSERT_FALSE( receivedPubAck );
+    }
+
+    return mqttStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+static bool connectToServerWithBackoffRetries( NetworkContext_t * pNetworkContext )
+{
+    bool isSuccessful = false;
+    RetryUtilsStatus_t retryUtilsStatus = RetryUtilsSuccess;
+    RetryUtilsParams_t reconnectParams;
+    TransportSocketStatus_t transportStatus = TRANSPORT_SOCKET_STATUS_SUCCESS;
+
+    /* Initializer server information. */
+    serverInfo.pHostName = BROKER_ENDPOINT;
+    serverInfo.hostNameLength = strlen( BROKER_ENDPOINT );
+    serverInfo.port = BROKER_PORT;
+
+    /* Initialize SocketsConfig. */
+    socketsConfig.enableTls = true;
+    socketsConfig.pAlpnProtos = NULL;
+    socketsConfig.maxFragmentLength = 0;
+    socketsConfig.disableSni = true;
```

Comment:
Please set to this false. We are connecting to AWS IoT.

Reply:
ok~